r/tests/testthat/test-dplyr-group-by.R

```diff
@@ -62,9 +62,23 @@ test_that("ungroup", {
       select(int, chr) %>%
       ungroup() %>%
       filter(int > 5) %>%
-      summarize(min_int = min(int)),
-    tbl,
-    warning = TRUE
+      collect(),
+    tbl
+  )
+
+  # to confirm that the above expectation is actually testing what we think it's
+  # testing, verify that expect_dplyr_equal() distinguishes between grouped and
+  # ungrouped tibbles
+  expect_error(
+    expect_dplyr_equal(
+      input %>%
+        group_by(chr) %>%
+        select(int, chr) %>%
+        (function(x) if (inherits(x, "tbl_df")) ungroup(x) else x) %>%
+        filter(int > 5) %>%
+        collect(),
+      tbl
+    )
```

Comment:
I'm really not sure about this as I can see compelling arguments both ways for including or not including this.  I'm a bit unsure as it feels like it's testing `expect_dplyr_equal()` and not `group_by()`, and given we don't have plans to revert the updated (attribute-testing) behaviour of `expect_dplyr_equal()` I don't know if it's strictly necessary. On the other hand, if something somehow goes wrong with that function, this would catch it.

Can we get someone else's input on this?

Reply:
I agree, this does feel a little strange to have this here, but I think it's worth keeping it because it's reasonably easy to follow when it appears here and it could conceivably prevent a future regression in the testing code.

Reply:
That makes sense, all good then!